src/logging.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Debug from 'debug'
+
+ /** @public */
+export interface Logger {
+  // tslint:disable-next-line:no-any automatically detect any inputs
+  debug(message?: any, ...optionalParams: any[]): void
+  // tslint:disable-next-line:no-any automatically detect any inputs
+  warn(message?: any, ...optionalParams: any[]): void
+  // tslint:disable-next-line:no-any automatically detect any inputs
+  info(message?: any, ...optionalParams: any[]): void
+  // tslint:disable-next-line:no-any automatically detect any inputs
+  error(message?: any, ...optionalParams: any[]): void
+  deprecate(feature: string, alternative: string): void
+}
+
+/** @hidden */
+class DefaultLogger implements Logger {
+  private logName = 'actions-on-google'
+  private debug_ = Debug(`${this.logName}:debug`)
```

Comment:
Can you change the naming style to use `_method` and `_property` instead?

For this library, we use that naming style for private methods [for example here](https://github.com/actions-on-google/actions-on-google-nodejs/blob/65045e6e46c2fb847d06585129fe3608b534227c/src/service/actionssdk/conversation/user.ts#L288).